spec/models/user_spec.rb

```diff
@@ -2799,6 +2800,29 @@ def reset_last_seen_cache!(user)
     end
   end
 
+  describe "#grouped_unread_high_priority_notifications" do
+    it "returns a map of high priority types to their unread count" do
+      Fabricate(:notification, user: user, notification_type: 1, high_priority: true, read: true)
+      Fabricate(:notification, user: user, notification_type: 1, high_priority: true, read: false)
+      Fabricate(:notification, user: user, notification_type: 1, high_priority: false, read: true)
+      Fabricate(:notification, user: user, notification_type: 1, high_priority: false, read: false)
+
+      Fabricate(:notification, user: user, notification_type: 2, high_priority: true, read: false, topic: nil)
+
+      Fabricate(:notification, user: user, notification_type: 3, high_priority: true, read: false).tap do |n|
+        n.topic.trash!(Fabricate(:admin))
+      end
+
+      Fabricate(:notification, user: user, notification_type: 3, high_priority: false, read: true)
+
+      Fabricate(:notification, notification_type: 4, high_priority: true, read: false)
+
+      expect(user.grouped_unread_high_priority_notifications).to contain_exactly(
+        *{ 1 => 1, 2 => 1 }.to_a
+      )
```

Comment:
```suggestion
      expect(user.grouped_unread_high_priority_notifications).to eq({1 => 1, 2 => 1})
```

Reply:
This is interesting, I thought that the keys order mattered when checking equality but it turned out it wasn't:

```
irb(main):001:0> { 1 => 1, 2 => 1 } == { 2 => 1, 1 => 1 }
=> true
```